zetaclient/chains/bitcoin/tx_script.go

```diff
@@ -245,3 +276,211 @@ func DecodeTSSVout(vout btcjson.Vout, receiverExpected string, chain chains.Chai
 
 	return receiverVout, amount, nil
 }
+
+// decodeInscriptionPayload checks the envelope for the script monitoring. The format is
+// OP_FALSE
+// OP_IF
+//
+//		OP_PUSHDATA_N ...
+//	 ...
+//		OP_PUSHDATA_N ...
+//
+// OP_ENDIF
+//
+// Note: the total data pushed will always be more than 80 bytes and within the btc transaction size limit.
+func decodeInscriptionPayload(t *scriptTokenizer) ([]byte, error) {
+	if !t.Next() || t.Opcode() != txscript.OP_FALSE {
+		return nil, fmt.Errorf("OP_FALSE not found")
+	}
+
+	if !t.Next() || t.Opcode() != txscript.OP_IF {
+		return nil, fmt.Errorf("OP_IF not found")
+	}
+
+	memo := make([]byte, 0)
+	var next byte
+	for {
+		if !t.Next() {
+			if t.Err() != nil {
+				return nil, t.Err()
+			}
+			return nil, fmt.Errorf("should contain more data, but script ended")
+		}
+
+		next = t.Opcode()
+
+		if next == txscript.OP_ENDIF {
+			break
+		}
+
+		if next < txscript.OP_DATA_1 || next > txscript.OP_PUSHDATA4 {
+			return nil, fmt.Errorf("expecting data push, found %d", next)
+		}
+
+		memo = append(memo, t.Data()...)
+	}
+
+	return memo, nil
+}
+
+// checkInscriptionEnvelope decodes the envelope for the script monitoring. The format is
+// OP_PUSHBYTES_32 <32 bytes> OP_CHECKSIG <Content>
+func checkInscriptionEnvelope(t *scriptTokenizer) error {
+	if !t.Next() || t.Opcode() != txscript.OP_DATA_32 {
+		return fmt.Errorf("cannot obtain public key bytes")
+	}
+
+	if !t.Next() || t.Opcode() != txscript.OP_CHECKSIG {
+		return fmt.Errorf("cannot parse OP_CHECKSIG")
+	}
+
+	return nil
+}
+
+func makeScriptTokenizer(script []byte) scriptTokenizer {
```

Comment:
```suggestion
func newScriptTokenizer(script []byte) scriptTokenizer {
```
I think new termninology is more adapted?